src/meta/meta_ckpt.c

```diff
@@ -510,6 +513,17 @@ __wt_meta_ckptlist_get(
         ckpt = &ckptbase[slot];
         ckpt->order = (slot == 0) ? 1 : ckptbase[slot - 1].order + 1;
         __wt_seconds(session, &ckpt->sec);
+        /*
+         * Update time value for most recent checkpoint, not letting it move backwards. It is
+         * possible to race here, so use atomic CAS. This code relies on the fact that anyone we
+         * race with will only increase (never decrease) the most recent checkpoint time value.
+         */
+        while (true) {
+            most_recent = conn->ckpt_most_recent;
+            if (ckpt->sec <= most_recent ||
+              __wt_atomic_cas64(&conn->ckpt_most_recent, most_recent, ckpt->sec))
+                break;
+        }
```

Comment:
There's no need for the local variable here, is there? I don't mind it, I guess.

Reply:
My thinking was that the local variable ensures that we are using the same value for `most_recent` in both parts of the or.  Otherwise two threads racing at the CAS could result in the one with the smaller time value (`ckpt->sec`) going last and overwriting the higher value.

Reply:
In an offline conversation with @keithbostic he pointed out that the compiler is allowed to elide the local variable. So my intent of ensuring that the same value is used on both sides of that or isn't guaranteed.

Reply:
I think it is important to use the same value for both pieces. We make reference to some of that in `include/column.i` and `include/log.i`. I feel like we need to do something here to make sure the same value is used in both the comparison and the atomic.

Reply:
Never mind! You were fixing it in the correct way as I was typing that!